### PR TITLE
Update MXQuant doc

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

documentation
## Description

transfer to AutoRound Quant

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Updated documentation for AutoRound Quantization API

- Added example using Hugging Face models

- Included code snippet for model quantization and inference

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["MXQuantConfig"] -- "updated to" --> B["AutoRoundConfig"]
  B -- "added example" --> C["Hugging Face models"]
  C -- "included code" --> D["Model quantization and inference"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>PT_MXQuant.md</strong><dd><code>Updated to AutoRound Quantization API</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/source/3x/PT_MXQuant.md

<ul><li>Updated API usage from MXQuant to AutoRound<br> <li> Added example with Hugging Face models<br> <li> Included code for quantization and inference</ul>

</details>

  </td>
  <td><a href="https://github.com/intel/neural-compressor/pull/2309/files#diff-212b1f1ca2f061a560946ee8e66ad935d66436a8126924525563857b44ab01e0">+30/-6</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by PRAgent4INC** [comment]
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 2 🔵🔵⚪⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/intel/neural-compressor/pull/2309/files#diff-212b1f1ca2f061a560946ee8e66ad935d66436a8126924525563857b44ab01e0R106-R106'><strong>Typo</strong></a>

There is a typo in the comment `# quantize the model and save to output_dir`. It should be `# quantize the model and save to output_dir`.
</summary>

```markdown
# quantize the model and save to output_dir
```

</details>

<details><summary><a href='https://github.com/intel/neural-compressor/pull/2309/files#diff-212b1f1ca2f061a560946ee8e66ad935d66436a8126924525563857b44ab01e0R122-R122'><strong>Example Link</strong></a>

The link provided in the examples section points to a non-existent path. Ensure the path is correct and the example exists.
</summary>

```markdown
- PyTorch [huggingface models](/examples/pytorch/multimodal-modeling/quantization/auto_round/llama4)
```

</details>

</td></tr>
</table>

**Comment by PRAgent4INC** [comment]
## PR Code Suggestions ✨



Explore these optional code suggestions:

<table><thead><tr><td><strong>Category</strong></td><td align=left><strong>Suggestion&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></td><td align=center><strong>Impact</strong></td></tr><tbody><tr><td rowspan=1>General</td>
<td>



<details><summary>Update API name consistency</summary>

___


**Correct the API name to match the description.**

[docs/source/3x/PT_MXQuant.md [86]](https://github.com/intel/neural-compressor/pull/2309/files#diff-212b1f1ca2f061a560946ee8e66ad935d66436a8126924525563857b44ab01e0R86-R86)

```diff
-To get a model quantized with Microscaling Data Types, users can use the Microscaling Quantization API as follows.
+To get a model quantized with AutoRound Data Types, users can use the AutoRound Quantization API as follows.
```
<details><summary>Suggestion importance[1-10]: 7</summary>

__

Why: The suggestion correctly updates the API name to match the description, improving clarity and consistency. However, it does not address a critical issue and offers a minor improvement.


</details></details></td><td align=center>Medium

</td></tr></tr></tbody></table>

**Comment by thuang6** [comment]
"It adapts a granularity falling between per-channel and per-tensor to balance accuracy and memory consumption." in introduction section looks not right.  block size 32 is normally smaller than channel dimension.  @mengniwang95, should we remove this sentence?

**Comment by thuang6** [comment]
Also formular "The exponent (exp) is equal to **torch.floor(torch.log2(amax))**" in introduction section is not right.  According to recipe document, the formular is: **clamp(floor(log2(amax)) - maxExp, -127, 127)**, Where maxExp is the largest power-of-two representable in the element data type, e.g. for element FP8 E4M3, maxExp is 8, FP4 E2M1, maxExp is 2.  @mengniwang95 , please double confirm if it is default option used in auto-round

File: docs/source/3x/PT_MXQuant.md [hunk 1]
```diff
@@ -83,19 +83,46 @@ The exponent (exp) is equal to torch.floor(torch.log2(amax)), MAX is the represe
...
+# inference
+text = "There is a girl who likes adventure,"
+inputs = tokenizer(text, return_tensors="pt").to(model.device)
+print(tokenizer.decode(model.generate(**inputs, max_new_tokens=10)[0]))
 ```
 
 ## Examples
 
-- PyTorch [huggingface models](/examples/pytorch/nlp/huggingface_models/language-modeling/quantization/mx_quant)
+- PyTorch [huggingface models](/examples/pytorch/multimodal-modeling/quantization/auto_round/llama4)
```
Comment (thuang6) [inline docs/source/3x/PT_MXQuant.md]: "huggingface models" is too general, better use "LLM/VLM models", later after Flux model example is ready, we can add "Diffusion models"

**Comment by mengniwang95** [comment]
> "It adapts a granularity falling between per-channel and per-tensor to balance accuracy and memory consumption." in introduction section looks not right. block size 32 is normally smaller than channel dimension. @mengniwang95, should we remove this sentence?

yes, you ar right

**Comment by mengniwang95** [comment]
> Also formular "The exponent (exp) is equal to **torch.floor(torch.log2(amax))**" in introduction section is not right. According to recipe document, the formular is: **clamp(floor(log2(amax)) - maxExp, -127, 127)**, Where maxExp is the largest power-of-two representable in the element data type, e.g. for element FP8 E4M3, maxExp is 8, FP4 E2M1, maxExp is 2. @mengniwang95 , please double confirm if it is default option used in auto-round

 clamp(floor(log2(amax)) - maxExp, -127, 127) is used in autoround

File: docs/source/3x/PT_MXQuant.md (hunk 1 above)
Comment (Kaihui-intel) [reply docs/source/3x/PT_MXQuant.md]: https://github.com/intel/neural-compressor/pull/2309/commits/b317f05df3ee5f1b90f038ac3e5e1f89713fc181

**Comment by thuang6** [comment]
> > Also formular "The exponent (exp) is equal to **torch.floor(torch.log2(amax))**" in introduction section is not right. According to recipe document, the formular is: **clamp(floor(log2(amax)) - maxExp, -127, 127)**, Where maxExp is the largest power-of-two representable in the element data type, e.g. for element FP8 E4M3, maxExp is 8, FP4 E2M1, maxExp is 2. @mengniwang95 , please double confirm if it is default option used in auto-round
> 
> clamp(floor(log2(amax)) - maxExp, -127, 127) is used in autoround

@Kaihui-intel , please help to update formular as well

**Comment by Kaihui-intel** [comment]
> > > Also formular "The exponent (exp) is equal to **torch.floor(torch.log2(amax))**" in introduction section is not right. According to recipe document, the formular is: **clamp(floor(log2(amax)) - maxExp, -127, 127)**, Where maxExp is the largest power-of-two representable in the element data type, e.g. for element FP8 E4M3, maxExp is 8, FP4 E2M1, maxExp is 2. @mengniwang95 , please double confirm if it is default option used in auto-round
> > 
> > 
> > clamp(floor(log2(amax)) - maxExp, -127, 127) is used in autoround
> 
> @Kaihui-intel , please help to update formular as well

https://github.com/intel/neural-compressor/pull/2309/commits/2d9c95de8344247d5a259c9095f92bc7d9cc9a9a
